tests/shell/__init__.py

```diff
@@ -85,54 +120,37 @@ class InteractiveShellType(type):
         },
         "csh": {
             "activator": "csh",
-            # Trying to use -x with `tcsh` on `macOS` results in some problems:
-            # This error from `PyCharm`:
-            # BrokenPipeError: [Errno 32] Broken pipe (writing to self.proc.stdin).
-            # .. and this one from the `macOS` terminal:
-            # pexpect.exceptions.EOF: End Of File (EOF).
-            # 'args': ('-x',),
-            "init_command": (
-                f'set _CONDA_EXE="{CONDA_PACKAGE_ROOT}/shell/bin/conda"; '
-                f"source {CONDA_PACKAGE_ROOT}/shell/etc/profile.d/conda.csh;"
-            ),
+            # "args": ("-v", "-x"),  # for debugging
+            # unset conda alias before calling conda shell hook
+            "init_command": f'unalias conda;\neval "`{EXE_UNIX} -m conda shell.csh hook {DEV_ARG}`"',
```

Comment:
We are forced to do this since the `conda.csh` script actually serves two different purposes. The first time it's called (i.e., `alias conda` is undefined) it defines the `conda` alias. Future calls are treated as actual `conda` commands.

I'm thinking we should solve this by splitting the `conda.csh` script into two scripts, a setup script that defines the alias and a `conda` "function" script.